src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -700,6 +700,32 @@ class IonRawBinaryWriter_1_1 internal constructor(
         hasFieldName = false
     }
 
+    override fun stepInEExp(systemMacro: SystemMacro) {
+        confirm(numAnnotations == 0) { "Cannot annotate an E-Expression" }
+
+        if (currentContainer.type == STRUCT && !hasFieldName) {
+            if (!currentContainer.usesFlexSym) switchCurrentStructToFlexSym()
+            buffer.writeByte(FlexInt.ZERO)
+            currentContainer.length++
+        }
```

Comment:
This could use a comment imo, both here and in other occurrences of the same fragment. What's happening here is that a macro may produce a field name. If it does, it must be an inline field name?
This is a strange corner of the spec I think.